### ISSUE TITLE
[Bug]: search results empyt, after restarting milvus standalone

### ISSUE THREAD

**Issue by snakecy** [milvus-io/milvus#17257]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Environment

```markdown
- Milvus version:
- Deployment mode(standalone or cluster): standalone
- SDK version(e.g. pymilvus v2.0.0rc2): pymilvus 2.0.2
- OS(Ubuntu or CentOS): ubuntu
- CPU/Memory: 8g
- GPU: 
- Others:
```

### Current Behavior

when i do search operator

### Milvus Log

[2022/05/27 06:06:55.315 +00:00] [DEBUG] [proxy_session_manager.go:108] ["success to send search result"] [node=248] [base="msg_type:SearchResult msgID:433485580341870728 timestamp:433489606155173891 sourceID:248 "]
[2022/05/27 06:06:55.315 +00:00] [DEBUG] [task.go:1963] ["Proxy Search PostExecute stage2"] [len(validSearchResults)=0]
[2022/05/27 06:06:55.315 +00:00] [DEBUG] [task.go:1966] ["Proxy Search PostExecute stage2 failed"] [filterReason="empty search result\n"]
[2022/05/27 06:06:55.315 +00:00] [DEBUG] [time_recorder.go:78] ["searchTask PostExecute: done (2ms)"]

[2022/05/27 06:06:57.737 +00:00] [INFO] [grpclog.go:37] ["[transport]transport: loopyWriter.run returning. connection error: desc = "transport is closing""]
[2022/05/27 06:07:17.890 +00:00] [WARN] [tso.go:177] ["clock offset is huge, check network latency and clock skew"] [jet-lag=162.1116ms] [prev-physical=2022/05/27 06:07:17.728 +00:00] [now=2022/05/27 06:07:17.890 +00:00]

**Comment by yanliang567** [comment]
@snakecy thank you for the issue. could you please use this [script](https://github.com/milvus-io/milvus/tree/master/deployments/export-log) to export Milvus logs

**Comment by snakecy** [comment]
> @snakecy thank you for the issue. could you please use this [script](https://github.com/milvus-io/milvus/tree/master/deployments/export-log) to export Milvus logs

the standalone log


[milvus.log](https://github.com/milvus-io/milvus/files/8792831/milvus.log)

**Comment by xiaofan-luan** [comment]
see couple of similar issues, mark it as critical urgent for now

**Comment by bigsheeper** [comment]
same as https://github.com/milvus-io/milvus/issues/16657

**Comment by bigsheeper** [comment]
QueryNode delete segments in defer func incorrectly
```
[32mmilvus-standalone |[0m [2022/05/29 06:50:39.632 +00:00] [DEBUG] [partition.go:64] ["remove a segment from replica"] [collectionID=433535115419713537] [partitionID=433535115616583681] [segmentID=433535119063515137]
[32mmilvus-standalone |[0m [2022/05/29 06:50:39.633 +00:00] [DEBUG] [segment.go:239] ["delete segment from memory"] [collectionID=433535115419713537] [partitionID=433535115616583681] [segmentID=433535119063515137]
[32mmilvus-standalone |[0m [2022/05/29 06:50:39.633 +00:00] [DEBUG] [partition.go:64] ["remove a segment from replica"] [collectionID=433535115419713537] [partitionID=433535115668750337] [segmentID=433535119364980737]
[32mmilvus-standalone |[0m [2022/05/29 06:50:39.633 +00:00] [DEBUG] [segment.go:239] ["delete segment from memory"] [collectionID=433535115419713537] [partitionID=433535115668750337] [segmentID=433535119364980737]
```
same as https://github.com/milvus-io/milvus/issues/16657

**Comment by bigsheeper** [comment]
This issue would be fixed in version 2.1 @snakecy

**Comment by xiaofan-luan** [comment]
close for now, wait for 2.1 release
